src/opd_loader.rs

```diff
@@ -1,52 +1,43 @@
-use std::future::Future;
-
 use crate::PointCloudAsset;
 use bevy::asset::LoadedAsset;
+use bevy::math::Vec3A;
 use bevy::render::render_resource::PrimitiveTopology;
 use bevy::{asset::AssetLoader, prelude::Mesh};
 
 pub struct OpdLoader;
 
 impl OpdLoader {
-    pub fn load_opd<'a>(bytes: &'a [u8]) -> impl Future<Output = PointCloudAsset> + 'a {
-        async move {
-            let file = opd_parser::parse(bytes).unwrap().1;
-            let mut positions = Vec::new();
-
-            let mut max_position = [f32::MIN, f32::MIN, f32::MIN];
-            let mut min_position = [f32::MAX, f32::MAX, f32::MAX];
-
-            for i in file.centroids.into_iter() {
-                let pos: [f32; 3] = i.offset.into();
-                for i in 0..3 {
-                    max_position[i] = max_position[i].max(pos[i]);
-                    min_position[i] = min_position[i].min(pos[i]);
-                }
-                positions.push(pos);
-            }
+    pub async fn load_opd<'a>(bytes: &'a [u8]) -> Result<PointCloudAsset, anyhow::Error> {
+        let file = opd_parser::parse(bytes).map_err(|e| e.to_owned())?.1;
+        let mut positions = Vec::new();
+
+        let mut max_position = Vec3A::new(f32::MIN, f32::MIN, f32::MIN);
+        let mut min_position = Vec3A::new(f32::MAX, f32::MAX, f32::MAX);
+
+        for i in file.centroids.into_iter() {
+            let pos: [f32; 3] = i.offset.into();
```

Comment:
I think `pos` and `positions` can also be `Vec3` or `Vec3A` because `Mesh::insert_attribute` works with those. It only requires `T: Into<VertexAttributeValues>`.